src/ir.cpp

```diff
@@ -18440,7 +18457,7 @@ static void buf_read_value_bytes(CodeGen *codegen, uint8_t *buf, ConstExprValue
         case TypeTableEntryIdVoid:
             return;
         case TypeTableEntryIdBool:
-            val->data.x_bool = (buf[0] != 0);
+            val->data.x_bool = (buf[0] != 0);  // NOLINT(clang-analyzer-core.UndefinedBinaryOperatorResult)
```

Comment:
Ignore warning for uninitialized value. Can occur if this is within uninitialized array (or at least that is my interpretation.